### PR TITLE
Add login action

### LINKED ISSUE #5: Create a login action
This should be run from a script named `exe/sdr`.  It should prompt the user for email, password, and then log them in and store the received token in `~/.sdr/token.json`

Example:
```
sdr login --host http://localhost:3000/
```

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?

Need a way to get credentials. Fixes #5 

## Was the documentation (README, wiki) updated?
yes

### CODE REVIEW

File: lib/sdr_client/credentials.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  # The stored credentials
+  class Credentials
+    # @param [String] a json string that contains a field 'token'
+    def self.write(body)
+      json = JSON.parse(body)
+      Dir.mkdir(credentials_path, 0o700) unless Dir.exist?(credentials_path)
+      File.open(credentials_file, 'w', 0o600) do |file|
+        file.write(json.fetch('token'))
+      end
+      puts 'Signed in.'
```
[the remark this answers is not in the document]
Comment (jcoyne) [reply lib/sdr_client/credentials.rb]: Maybe. I don't know enough about how we will do that in the future. For now, this is a CLI app.

[the remark this answers is not in the document]
Comment (justinlittman) [reply lib/sdr_client/credentials.rb]: Won't we likely be using it for Google Books as a library?

[the remark this answers is not in the document]
Comment (mjgiarlo) [reply lib/sdr_client/credentials.rb]: We think the Google Books app will use sdr-client as a library eventually, right? That's no reason to walk away from CLI patterns in this PR... but may be a reason to write up an issue for making sure we don't bake CLI interactions into the library in the future. Shall I write up a new issue and leave it in the backlog for now? @jcoyne @justinlittman

[the remark this answers is not in the document]
Comment (jcoyne) [reply lib/sdr_client/credentials.rb]: It seems to me we already have this ticket for that: https://github.com/sul-dlss/google-books/issues/56

[the remark this answers is not in the document]
Comment (mjgiarlo) [reply lib/sdr_client/credentials.rb]: I tossed a comment at that issue:

> This work includes using the sdr-client to access the SDR API, and may involved modifying sdr-client to disentangle CLI concerns from library concerns (e.g., `puts`ing all over the place).

[the remark this answers is not in the document]
Comment (mjgiarlo) [reply lib/sdr_client/credentials.rb]: Whaddya say, @justinlittman?

**Comment by mjgiarlo** [comment]
Merging. @justinlittman we'll take any disentangling as part of work on the issue linked above.

**Pull request merged**